### PR TITLE
doc: add signed_by_b0_*_image.bin to output files

### PR DESCRIPTION
**Pull request by greg-fer**
Added entries to for MCUboot output file list:
signed_by_b0_s0_image.bin and signed_by_b0_s1_image.bin. NCSDK-19379.

----

Related to https://devzone.nordicsemi.com/f/nordic-q-a/95700/signed_by_b0_s1_image-bin-vs-signed_by_mcuboot_and_b0_s1_image_update-bin

### CODE REVIEW

**Review by annwoj — APPROVED** [review]

**Review by gregersrygg — CHANGES_REQUESTED** [review]
The original question from the customer is not being answered here. What is the difference between these files and those that are also signed by mcuboot? I don't think CMakeLists.txt and the cmake files are relevant, but it could be mentioned that these files are only intermediate images when also using MCUboot. These images can be used when using NSIB without mcuboot (but then it doesn't belong in this mcuboot table).

**Comment by greg-fer** [comment]
@gregersrygg , I've updated the phrasing and moved the files to other output files. Please check again.

File: doc/nrf/app_dev/build_and_config_system/index.rst [hunk 1]
```diff
@@ -173,9 +173,9 @@ MCUboot output build files
...
 +------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
-| :file:`signed_by_mcuboot_and_b0_s0_image_update.bin`             | MCUboot update for s0.                                                                                                                                                                                                                                                       |
+| :file:`signed_by_mcuboot_and_b0_s0_image_update.bin`             | MCUboot update for s0. Created using the :file:`signed_by_b0_s0_image.bin` (see :ref:`app_build_output_files_other`).                                                                                                                                                        |
```
Comment (gregersrygg) [inline doc/nrf/app_dev/build_and_config_system/index.rst:176]:
```suggestion
| :file:`signed_by_mcuboot_and_b0_s0_image_update.bin`             | MCUboot update image for s0 signed by both MCUboot and NSIB. The MCUboot signature is used by MCUboot to verify the integrity of the image before swapping and the NSIB signature is used by NSIB before booting the image.                                                                                                                                                        |
```

File: doc/nrf/app_dev/build_and_config_system/index.rst [hunk 2]
```diff
@@ -173,9 +173,9 @@ MCUboot output build files
...
 +------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
-| :file:`signed_by_mcuboot_and_b0_s1_image_update.bin`             | MCUboot update for s1.                                                                                                                                                                                                                                                       |
+| :file:`signed_by_mcuboot_and_b0_s1_image_update.bin`             | MCUboot update for s1. Created using the :file:`signed_by_b0_s1_image.bin` (see :ref:`app_build_output_files_other`).                                                                                                                                                        |
```
Comment (gregersrygg) [inline doc/nrf/app_dev/build_and_config_system/index.rst:178]:
```suggestion
| :file:`signed_by_mcuboot_and_b0_s1_image_update.bin`             | MCUboot update image for s1 signed by both MCUboot and NSIB. The MCUboot signature is used by MCUboot to verify the integrity of the image before swapping and the NSIB signature is used by NSIB before booting the image                  |
```

File: doc/nrf/app_dev/build_and_config_system/index.rst [hunk 3]
```diff
@@ -204,18 +204,24 @@ Other output build files
...
+| :file:`signed_by_b0_s0_image.bin` | Intermediate file generated by :file:`sign_cmake` and used for :ref:`bootloader`.                                                 |                                                                                                                                                                                             |
+|                                   | Can be used to generate the :file:`signed_by_mcuboot_and_b0_s0_image_update.bin` when using NSIB with MCUboot.                    |
++-----------------------------------+-----------------------------------------------------------------------------------------------------------------------------------+
+| :file:`signed_by_b0_s1_image.bin` | Intermediate file generated by :file:`sign_cmake` and used for :ref:`bootloader`.                                                 |
+|                                   | Can be used to generate the :file:`signed_by_mcuboot_and_b0_s1_image_update.bin` when using NSIB with MCUboot.                    |
```
Comment (gregersrygg) [inline doc/nrf/app_dev/build_and_config_system/index.rst:226]:
```suggestion
| :file:`signed_by_b0_s0_image.bin` | Intermediate file only signed by NSIB.                                                 |                                                                                                                                                                                             |
|                                   |                    |
+-----------------------------------+-----------------------------------------------------------------------------------------------------------------------------------+
| :file:`signed_by_b0_s1_image.bin` | Intermediate file only signed by NSIB.                                                 |
|                                   |                    |
```

I struggle to see the usefulness of describing these files here except that they're just intermediate files. Is it even possible to use NSIB without MCUboot? Would you have the application as S0/S1 then? Or would you need some other bootloader? This sounds very edge-case to me, but please correct me if there is an actual use-case for this.
The way I understood the question from the customer was why we had the two set of files and the difference between them. There was also some confusion in the case w.r.t. swapping of S0/S1 and how the mcuboot signature was being used. So I think it's more important to explain why the image needs to be signed by both NSIB and MCUboot.

**Comment by greg-fer** [comment]
@gregersrygg , @aHaugl , please have another look. Thanks!

**Review by gregersrygg — APPROVED** [review]
Thanks :)

**Review by aHaugl — APPROVED** [review]

**Pull request merged**